internal/api/utils.go

```diff
@@ -217,24 +210,47 @@ func truncateComponentUnits(recommendationJSON map[string]interface{}) map[strin
 		}
 		return value
 	}
+	var convertedValueCPU float64
 
-	truncateCPU := func(cpu map[string]interface{}) bool {
-		cpuInCores, ok := cpu["amount"].(float64)
-		if ok {
-			cpu["amount"] = truncateToThreeDecimalPlaces(cpuInCores)
-			cpu["format"] = nil
-		}
-		return ok
+	if cpuUnit == "millicores" {
+		convertedValueCPU = math.Round(cpuValue * 1000) // millicore values don't require decimal precision
+	} else if cpuUnit == "cores" {
+		convertedValueCPU = truncateToThreeDecimalPlaces(cpuValue)
+	} else {
+		convertedValueCPU = cpuValue
```

Comment:
Nit and not a blocker, considering our default we could have checked for cpuUnit == "cores" first, just to save one comparison